src/oal_agent/cli.py

```
@@ -65,5 +67,38 @@ def analyze(contract_file: str):
+    except requests.exceptions.HTTPError as e:
+        if e.response.status_code == 404:
+            click.echo(f"Error: Job with ID '{job_id}' not found.")
+        else:
+            click.echo(f"HTTP error occurred: {e}")
+    except requests.exceptions.ConnectionError as e:
+        click.echo(f"Error: Could not connect to the API server. Is it running? ({e})")
+    except requests.exceptions.Timeout as e:
+        click.echo(f"Error: The request timed out: {e}")
+    except requests.exceptions.RequestException as e:
+        click.echo(f"An unexpected error occurred: {e}")
+    except Exception as e:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)